pkg/ingress/kube/ingress/controller.go

```diff
@@ -526,46 +525,43 @@ func (c *controller) ConvertHTTPRoute(convertOptions *common.ConvertOptions, wra
 
 		wrapperHttpRoutes := make([]*common.WrapperHTTPRoute, 0, len(rule.HTTP.Paths))
 		for _, httpPath := range rule.HTTP.Paths {
+			var pathType common.PathType
+			if wrapper.AnnotationsConfig.NeedRegexMatch() {
```

Comment:
Here we can determine the value of original path.

Reply:
get, so I should convert the original path to httpMatch here and remove generateHttpMatches func. then we don't need to tarverse http matchs below.